app/src/components/tables/BlockTable/BlocksTable.tsx

```diff
@@ -29,37 +38,57 @@ const rowCountSelectOptions: SelectOptions[] | null = [
   { value: '20', label: '20 rows' },
 ];
 
-interface BlocksTableProps {
-  readonly total?: number;
-  readonly blocks: ApiData.Block[];
-  readonly showValidators?: boolean;
-  isTableLoading: boolean;
-  onSortingChange?: OnChangeFn<SortingState>;
-  sorting?: SortingState;
-  initialSorting?: SortingState;
-  setIsTableLoading: React.Dispatch<React.SetStateAction<boolean>>;
-}
+const initialSorting: SortingState = [
+  {
+    id: 'height',
+    desc: true,
+  },
+];
 
-export const BlocksTable: React.FC<BlocksTableProps> = ({
-  total,
-  blocks,
-  showValidators,
-  isTableLoading,
-  setIsTableLoading,
-  ...props
-}) => {
+export const BlocksTable: React.FC = () => {
   const { t } = useTranslation();
 
+  const [isTableLoading, setIsTableLoading] = useState(false);
   const [currentTime, setCurrentTime] = useState(Date.now());
   const [showTimestamp, setShowTimestamp] = useState(false);
 
-  const blocksTableOptions = useAppSelector(getBlocksTableOptions);
+  const dispatch = useAppDispatch();
+
+  const blocks = useAppSelector(getBlocks);
   const totalBlocks = useAppSelector(getTotalBlocks);
+  const latestBlock = useAppSelector(getLatestBlock);
+  const blockLoadingStatus = useAppSelector(getBlocksLoadingStatus);
+  const blocksTableOptions = useAppSelector(getBlocksTableOptions);
 
   const totalPages = useMemo(() => {
     return Math.ceil(totalBlocks / blocksTableOptions.pagination.pageSize);
   }, [blocksTableOptions, totalBlocks]);
 
+  const isLoadingPage =
+    blockLoadingStatus !== Loading.Complete && !blocks.length;
+
+  useEffect(() => {
+    dispatch(setInitialStateFromUrlSearchParams());
+  }, [dispatch]);
+
+  useEffect(() => {
+    // updated from WS
+    if (latestBlock) {
+      dispatch(updateBlocksWithLatest({ latestBlock }));
+    }
+  }, [latestBlock, dispatch]);
+
+  useEffect(() => {
+    dispatch(fetchBlocks(blocksTableOptions));
+  }, [dispatch, blocksTableOptions]);
+
+  useEffect(() => {
+    if (isTableLoading || isLoadingPage) {
+      setIsTableLoading(false);
+    }
```

Comment:
Just so I'm understanding this correctly.  IF the pointer to the blocks variable is updated (I.E. if blocks ever gets updated) then we want to check if we need to set the internal loading state to false.  Is there ever a chance where both of the conditionals may be false before this is hit?

Reply:
No, not that I can think of.

Reply:
Okay sound good!